# Step 1. Import the necessary libraries

In [107]:
import pandas as pd
import numpy as np
import seaborn as sb
import missingno as mo
import matplotlib.pyplot as plt
import warnings 
warnings.filterwarnings('ignore')
%matplotlib Inline

# Step 2. Import the dataset from this address. and assign it to a variable chipo

In [108]:
chipo= pd.read_csv('https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv' ,sep='\t')

In [109]:
chipo.tail()

order_id  quantity           item_name  \
4617      1833         1       Steak Burrito   
4618      1833         1       Steak Burrito   
4619      1834         1  Chicken Salad Bowl   
4620      1834         1  Chicken Salad Bowl   
4621      1834         1  Chicken Salad Bowl   

                                     choice_description item_price  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75

# Step 4. See the first 10 entries

In [110]:
chipo.head(10)

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   
5         3         1                           Chicken Bowl   
6         3         1                          Side of Chips   
7         4         1                          Steak Burrito   
8         4         1                       Steak Soft Tacos   
9         5         1                          Steak Burrito   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                NaN     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

# Step 5 What is the number of observations in the dataset?¶

In [111]:
chipo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


In [112]:
chipo.describe()

order_id     quantity
count  4622.000000  4622.000000
mean    927.254868     1.075725
std     528.890796     0.410186
min       1.000000     1.000000
25%     477.250000     1.000000
50%     926.000000     1.000000
75%    1393.000000     1.000000
max    1834.000000    15.000000

# Step 6. What is the number of columns in the dataset?

In [113]:
chipo.shape

(4622, 5)

# The Shape of the Dataframe Tell us That there are 4622 rows and % columns

# Step 7. Print the name of all the columns.¶

In [114]:
chipo.columns

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

# Step 8. How is the dataset indexed?

# Step 9. Which was the most-ordered item?

In [115]:
chipo.item_name.value_counts()

Chicken Bowl                             726
Chicken Burrito                          553
Chips and Guacamole                      479
Steak Burrito                            368
Canned Soft Drink                        301
Chips                                    211
Steak Bowl                               211
Bottled Water                            162
Chicken Soft Tacos                       115
Chips and Fresh Tomato Salsa             110
Chicken Salad Bowl                       110
Canned Soda                              104
Side of Chips                            101
Veggie Burrito                            95
Barbacoa Burrito                          91
Veggie Bowl                               85
Carnitas Bowl                             68
Barbacoa Bowl                             66
Carnitas Burrito                          59
Steak Soft Tacos                          55
6 Pack Soft Drink                         54
Chips and Tomatillo Red Chili Salsa       48
Chicken Cr

# Chicken Bowl Was Ordered The Most

# Step 10. For the most-ordered item, how many items were ordered?¶

In [116]:
chipo.item_name.value_counts()

Chicken Bowl                             726
Chicken Burrito                          553
Chips and Guacamole                      479
Steak Burrito                            368
Canned Soft Drink                        301
Chips                                    211
Steak Bowl                               211
Bottled Water                            162
Chicken Soft Tacos                       115
Chips and Fresh Tomato Salsa             110
Chicken Salad Bowl                       110
Canned Soda                              104
Side of Chips                            101
Veggie Burrito                            95
Barbacoa Burrito                          91
Veggie Bowl                               85
Carnitas Bowl                             68
Barbacoa Bowl                             66
Carnitas Burrito                          59
Steak Soft Tacos                          55
6 Pack Soft Drink                         54
Chips and Tomatillo Red Chili Salsa       48
Chicken Cr

# Step 11. What was the most ordered item in the choice_description column?

In [117]:
chipo.choice_description.value_counts()

[Diet Coke]                                                                                                134
[Coke]                                                                                                     123
[Sprite]                                                                                                    77
[Fresh Tomato Salsa, [Rice, Black Beans, Cheese, Sour Cream, Lettuce]]                                      42
[Fresh Tomato Salsa, [Rice, Black Beans, Cheese, Sour Cream, Guacamole, Lettuce]]                           40
                                                                                                          ... 
[[Tomatillo-Red Chili Salsa (Hot), Roasted Chili Corn Salsa (Medium)], [Guacamole, Cheese, Sour Cream]]      1
[Roasted Chili Corn Salsa, [Rice, Pinto Beans, Lettuce]]                                                     1
[Fresh Tomato (Mild), [Guacamole, Lettuce, Rice, Fajita Veggies, Sour Cream, Cheese]]                        1
[

# Step 12. How many items were orderd in total?

In [118]:
chipo.quantity.sum()

4972

# Step 13. Turn the item price into a float

In [119]:
chipo['item_price']=chipo['item_price'].str.strip(' ')
dollarizer = lambda x: float(x[1:-1])
chipo['item_price']=chipo['item_price'].apply(dollarizer)

In [120]:
chipo.item_price.dtype

dtype('float64')

# Step 14. How much was the revenue for the period in the dataset?

In [121]:
Revenue_Generated=(chipo['quantity'] * chipo['item_price']).sum()
print("The Revenue Generated For the period was $",str(np.round(Revenue_Generated,2)))

The Revenue Generated For the period was $ 38959.4


# Step 15. How many orders were made in the period?¶

In [136]:
Order_made=chipo['order_id'].value_counts().count()
print("The Number of the Order made in the Period Generated For the period was a count of ",Order_made)

The Number of the Order made in the Period Generated For the period was a count of  1834


# Step 16. What is the average revenue amount per order?

In [143]:
chipo['Revenue_Generated']=(chipo['quantity'] * chipo['item_price'])
chipo.groupby(by=['order_id']).sum().mean()['Revenue_Generated']

21.242857142857158

#  Step 17. How many different items are sold?

In [145]:
chipo['item_name'].value_counts().count()

50